main.js

```diff
@@ -44,91 +44,103 @@ export async function signCapabilityInvocation({
 
-  if(json && !('digest' in signed)) {
-    // compute digest for json
-    signed.digest = await createHeaderValue({data: json, useMultihash: true});
+    if(json && !('digest' in signed)) {
+      // compute digest for json
+      signed.digest = await createHeaderValue({data: json, useMultihash: true});
 
-    if(!('content-type' in signed)) {
-      signed['content-type'] = 'application/json';
+      if(!('content-type' in signed)) {
+        signed['content-type'] = 'application/json';
+      }
     }
-  }
-  // convert dates to unix time stamp
-  if(created instanceof Date) {
-    created = Math.floor(created.getTime() / 1000);
-  }
-  // set expiration 10 minutes into the future
-  expires = expires || Number.parseInt(created) + 600;
+    // convert dates to unix time stamp
+    if(created instanceof Date) {
+      created = Math.floor(created.getTime() / 1000);
+    }
+    // set expiration 10 minutes into the future
+    expires = expires || Number.parseInt(created) + 600;
 
-  // sign header
-  const {id: keyId} = invocationSigner;
-  const includeHeaders = [
-    '(key-id)', '(created)', '(expires)', '(request-target)',
-    'host', 'capability-invocation'];
-  if(json) {
-    includeHeaders.push('content-type');
-    includeHeaders.push('digest');
-  }
-  const plaintext = createSignatureString({
-    includeHeaders,
-    requestOptions: {url, method, headers: signed, created, expires, keyId}
-  });
-  const data = new TextEncoder().encode(plaintext);
-  const signature = base64Encode(await invocationSigner.sign({data}));
+    // sign header
+    const {id: keyId} = invocationSigner;
+    const includeHeaders = [
+      '(key-id)', '(created)', '(expires)', '(request-target)',
+      'host', 'capability-invocation'];
+    if(json) {
+      includeHeaders.push('content-type');
+      includeHeaders.push('digest');
+    }
+    const plaintext = createSignatureString({
+      includeHeaders,
+      requestOptions: {url, method, headers: signed, created, expires, keyId}
+    });
+    const data = new TextEncoder().encode(plaintext);
+    const signature = base64Encode(await invocationSigner.sign({data}));
 
-  signed.authorization = createAuthzHeader({
-    includeHeaders,
-    keyId,
-    signature,
-    created,
-    expires
-  });
+    signed.authorization = createAuthzHeader({
+      includeHeaders,
+      keyId,
+      signature,
+      created,
+      expires
+    });
 
-  // detect browser environment
-  if(isBrowser) {
-    // remove `host` header as it will be automatically set by the browser
-    delete signed.host;
-  }
+    // detect browser environment
+    if(isBrowser) {
+      // remove `host` header as it will be automatically set by the browser
+      delete signed.host;
+    }
 
-  return signed;
+    return signed;
+  } catch(cause) {
+    const error = new Error(
+      `Error invoking zCap for ${method.toUpperCase()} "${url}", ` +
+      `action: "${capabilityAction}".`);
```

Comment:
What do you think of this to avoid transforms on `method`? I considered `String(method).toUpperCase()`, but I really think we shouldn't transform it -- and we should just call out each key+value pair separately. We usually do this sort of thing as a details object vs. putting it into the string but I can see benefit here since these get pretty deep into stacks. Then again, with better `.cause` tooling in browsers, etc. moving forward we may not need to try and stuff all this info into the string.

```suggestion
    const error = new Error(
      'Error signing capability invocation.\n' +
      `method: "${method}",\n`,
      `url: "${url}"\n,` +
      `action: "${capabilityAction}"\n`);
```

Reply:
Sure, that works! Committing.